### PR TITLE
Add `ERC721ABI` and fix other abis

### LINKED ISSUE #739: Add missing `ERC721ABI` trait

### PR DESCRIPTION
**Pull request by andrew-fleming**
Supersedes #741 and fixes #739.

This PR proposes to normalize the interface/impl function order. Furthermore, this PR also proposes to reorder the internal functions to better match the order of their external counterparts

### CODE REVIEW

File: src/account/account.cairo [hunk 1]
```diff
@@ -154,24 +166,14 @@ mod Account {
...
-
     //
     // Internal
     //
 
     #[generate_trait]
     impl InternalImpl of InternalTrait {
         fn initializer(ref self: ContractState, _public_key: felt252) {
-            let mut unsafe_state = src5_state();
+            let mut unsafe_state = SRC5::unsafe_new_contract_state();
```
Comment (ericnordelo) [inline src/account/account.cairo:176]: Should we keep this? It should be removed when migrating to component anyway, buy I think src5_state looks better.

**Review by ericnordelo — COMMENTED** [review]
Small comment, but besides I think it looks good!

File: src/account/account.cairo (hunk 1 above)
Comment (andrew-fleming) [reply src/account/account.cairo:176]: Agreed! I think this got mixed up in the merge/release. Good catch

**Comment by andrew-fleming** [comment]
@ericnordelo I included the camelCase methods in the Accounts api (as mentioned [here](https://github.com/OpenZeppelin/cairo-contracts/pull/689#discussion_r1322315725)). There's a few camel `impl`s in the Account mod. Listing each camel impl seems a bit noisy IMO. I propose we group them as `CAMELCASE-SUPPORT` in the api. Wdyt?
Screen Shot 2023-10-03 at 12 50 39 PM

**Comment by ericnordelo** [comment]
It looks way better to me than adding each impl. I agree.

File: docs/modules/ROOT/pages/api/account.adoc [hunk 2]
```diff
@@ -98,10 +97,17 @@ Account contract implementation extending xref:ISRC6[`ISRC6`].
...
 * xref:#Account-\\__validate_declare__[`++__validate_declare__(self, class_hash)++`]
 
-.PublicKeyImpl
-
+.Non-standard
+* xref:#Account-\\__validate_deploy__[`++__validate_deploy__(self, hash, signature)++`]
 * xref:#Account-set_public_key[`++set_public_key(self, new_public_key)++`]
 * xref:#Account-get_public_key[`++get_public_key(self)++`]
+
+.camelcase-support
```
Comment (ericnordelo) [inline docs/modules/ROOT/pages/api/account.adoc:105]:
```suggestion
.camelCase-support
```

for consistency, but not needed at all. TOC comment, lol

**Review by ericnordelo — APPROVED** [review]
LGTM! Left a really small suggestion, and we need to add the camelCase support to other modules as well, but we can leave that for another PR.

**Comment by andrew-fleming** [comment]
@ericnordelo I updated the Account module so it's exposing external `impl`s (not `non-standard` fns). I didn't update the ERC20 `Non-Standard` name because I think this is enough for this PR and we'll have to change some things around anyway with the component migration. We could also just apply these changes in the migration PRs instead. Let me know what you think

File: src/token/erc20/interface.cairo [hunk 3]
```diff
@@ -54,25 +59,20 @@ trait ERC20ABI<TState> {
...
     ) -> bool;
-}
 
-#[starknet::interface]
-trait ERC20CamelABI<TState> {
-    fn name(self: @TState) -> felt252;
-    fn symbol(self: @TState) -> felt252;
-    fn decimals(self: @TState) -> u8;
+    // Camel case compatibility
+    // See https://docs.openzeppelin.com/contracts-cairo/0.7.0/interfaces#dual_interfaces
```
Comment (martriay) [inline src/token/erc20/interface.cairo:70]:
```suggestion
    // See https://docs.openzeppelin.com/contracts-cairo/0.8.0-beta.0/interfaces#dual_interfaces
```

File: src/token/erc721/interface.cairo [hunk 4]
```diff
@@ -87,3 +87,54 @@ trait IERC721ReceiverCamel<TState> {
...
+    // ISRC5
+    fn supports_interface(self: @TState, interface_id: felt252) -> bool;
+
+    // IERC721Metadata
+    fn name(self: @TState) -> felt252;
+    fn symbol(self: @TState) -> felt252;
+    fn token_uri(self: @TState, token_id: u256) -> felt252;
+
+    // Camel case compatibility
+    // See https://docs.openzeppelin.com/contracts-cairo/0.7.0/interfaces#dual_interfaces
```
Comment (martriay) [inline src/token/erc721/interface.cairo:124]:
```suggestion
    // See https://docs.openzeppelin.com/contracts-cairo/0.8.0-beta.0/interfaces#dual_interfaces
```

**Review by martriay — COMMENTED** [review]
need to fix conflicts and update to latest, other than that it looks good to me!

**Review by martriay — APPROVED** [review]

**Pull request merged**